selfie.c

```diff
@@ -1508,24 +1508,36 @@ int atoi(int* s) {
       // c was not a decimal digit
       return -1;
 
-    // assert: s contains a decimal number, that is, with base 10
-    n = n * 10 + c;
+    // assert: s contains a decimal number
+
+    // use base 10 but avoid integer overflow
+    if (n < INT_MAX / 10)
+      n = n * 10 + c;
+    else if (n == INT_MAX / 10) {
+      // In decimal representation of integers,
```

Comment:
The numerical value of INT_MIN is always greater by one than of INT_MAX but not necessarily the least signnificant digit. Try to find a counterexample for some number of bits other than 32. Anyway, please remove the comment and resubmit the PR.